tools/HiveToScripts/Program.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Concurrent;
+using System.Text.RegularExpressions;
+
+bool doLog = true;
+
+if (args.Intersect(["--help", "-h", "help", "?"]).Any())
+{
+    Console.WriteLine(
+@"Transforms hive test execution output into sh/ps1/http scripts with big inputs dumped as separate files:
+
+    ./HiveToScripts.exe [<path-to-hive-logs> [<output-path> [<json-rpc-url>]]]
+
+Example:
+
+    ./HiveToScripts.exe T:\logs T:\output http://localhost:8551
+");
+    return;
+}
+
+string hiveLogsDirectory = args.Length > 0 ? args[0] : "logs";
+
+if (Directory.Exists(Path.Combine(hiveLogsDirectory, "logs")))
+{
+    hiveLogsDirectory = Path.Combine(hiveLogsDirectory, "logs");
+}
+
+string outputDirectory = args.Length > 1 ? args[1] : "output";
+string jsonrpcUrl = args.Length > 2 ? args[2] : "http://localhost:8551";
+
+string[] simulatorFilePaths = Directory.GetFiles(hiveLogsDirectory, "*-simulator-*.log").Select(f => Path.Combine(outputDirectory, f)).ToArray();
+ConcurrentDictionary<string, TextWriter> fileDescriptors = new();
+
+foreach (string simulatorFilePath in simulatorFilePaths)
+{
+    using StreamReader simulatorFile = File.OpenText(simulatorFilePath);
+
+    if (Directory.Exists(outputDirectory))
+    {
+        foreach (string f in Directory.EnumerateFiles(outputDirectory))
+        {
+            File.Delete(f);
+        }
+    }
+    Directory.CreateDirectory(outputDirectory);
+    if (doLog) Console.WriteLine($"Running...");
+    HashSet<string> createdFiles = [];
+
+    using StreamWriter testListWriter = new(Path.Combine(outputDirectory, "__tests.txt"));
+    HashSet<string> reportedContainers = [];
+
+    Regex starter = new("^>>\\s+\\(([a-f0-9]+)\\)\\s+");
```

Comment:
Not all hive tests will write the container id like this.

Reply:
Which ones, could you share example? I've never faced such problem

Reply:
> Which ones, could you share example? I've never faced such problem

I know rpccompat  for one, i am not sure if there are more

Reply:
Fixed